src/main/webapp/app/shared/util/core-submission-shared/core-submission-utils.ts

```diff
@@ -1,33 +1,42 @@
 import { Review } from 'app/shared/model/firebase/firebase.model';
 
-export function useLastReviewedOnly<T>(obj: T): T | undefined {
-  function useLastReviewedOnlyRec(curObj: T): T | undefined {
+function shouldProtect(valuePath: string | undefined, path: string) {
+  return valuePath !== undefined && path.length > 0 && (path.startsWith(valuePath) || valuePath.startsWith(path));
+}
+
+export function useLastReviewedOnly<T>(obj: T, valuePath?: string): T | undefined {
+  function useLastReviewedOnlyRec(curObj: T, parentPath: string): T | undefined {
     const newObj: T = {} as T;
     for (const [key, value] of Object.entries(curObj as Record<string, unknown>)) {
+      const currentPath = `${parentPath}${parentPath.length > 0 ? '/' : ''}${key}`;
+      const protectCurrent = shouldProtect(valuePath, currentPath);
+      const protectParent = shouldProtect(valuePath, parentPath);
```

Comment:
Need to protect the whole object if the parent is in the valuePath since we don't want to drop the whole object.